File: service/javadsl/server/src/main/scala/com/lightbend/lagom/internal/javadsl/server/AdditionalRouters.scala
/*
 * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
 */
package com.lightbend.lagom.internal.javadsl.server
import java.util
import java.util.Collections

import play.routing.Router

trait AdditionalRouters {
  def getRouters: util.List[Router]
}

class EmptyAdditionalRouters extends AdditionalRouters {
  override def getRouters: util.List[Router] = Collections.emptyList()
}
class NonEmptyAdditionalRouters(val routers: util.List[Router]) extends AdditionalRouters {
  override def getRouters: util.List[Router] = routers
}

File: service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/AdditionalRouters.java
/*
 * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package com.lightbend.lagom.javadsl.server;

import play.routing.Router;

import java.util.Collections;
import java.util.List;

public interface AdditionalRouters {
    public static final AdditionalRouters EMPTY = new AdditionalRouters() {
        @Override
        public List<Router> getRouters() {
            return Collections.emptyList();
        }
    };

    List<Router> getRouters();
}

